lane.py

```diff
@@ -1306,364 +1215,127 @@ def search(self, query, search_client, pagination=None):
-        the works specific to the QueryGeneratedLane
+        qu = self.apply_age_range_filter(_db, qu, work_model)
+        qu, child_distinct = self.apply_customlist_filter(
+            qu, work_model, featured
+        )
+        return qu, (parent_distinct or child_distinct)
 
-        :return: query or None
+    def apply_age_range_filter(self, _db, qu, work_model):
+        """Filter out all books that are not classified as suitable for this
+        Lane's age range.
         """
-        # Only show works that can be borrowed or reserved.
-        qu = self.only_show_ready_deliverable_works(qu, work_model)
-
-        # Only show works for the proper audiences.
-        if self.audiences:
-            qu = qu.filter(work_model.audience.in_(self.audiences))
-
-        # Only show works in the source language.
-        if self.languages:
-            qu = qu.filter(edition_model.language.in_(self.languages))
-
-        # Add lane-specific details to query and return the result.
-        qu = self.lane_query_hook(qu, work_model=work_model)
-        if not qu:
-            # The hook may return None.
-            return None
-
-        if facets:
-            qu = facets.apply(self._db, qu, work_model, edition_model)
-
-        if pagination:
-            qu = pagination.apply(qu)
-
-        return qu
-
-    def featured_works(self, use_materialized_works=True):
-        """Find a random sample of books for the feed"""
-
-        # Lane.featured_works searches for books along a variety of facets.
-        # Because WorkBasedLanes are created for individual works as
-        # needed (instead of at app start), we need to avoid the relative
-        # slowness of those queries.
-        #
-        # We'll just ignore facets and return whatever we find.
-        if not use_materialized_works:
-            query = self.works()
+        if self.target_age == None:
+            return qu
+            
+        if (Classifier.AUDIENCE_ADULT in self.audiences
+            or Classifier.AUDIENCE_ADULTS_ONLY in self.audiences):
+            # Books for adults don't have target ages. If we're including
+            # books for adults, allow the target age to be empty.
+            audience_has_no_target_age = work_model.target_age == None
         else:
-            query = self.materialized_works()
-        if not query:
-            return []
-
-        target_size = self.library.featured_lane_size
-        return self.randomized_sample_works(
-            query, target_size=target_size, use_min_size=True
+            audience_has_no_target_age = False
+
+        # The lane's target age is an inclusive NumericRange --
+        # set_target_age makes sure of that. The work's target age
+        # must overlap that of the lane.
+        qu = qu.filter(
+            or_(
+                work_model.target_age.overlaps(self.target_age),
+                audience_has_no_target_age
+            )
```

Comment:
What happens if the audience is YA or children, but there's no target age?

Reply:
Nothing happens in `apply_age_range_filter` because it returns early if `self.target_age == None`. But in `WorkList.apply_audience_filter` the query is restricted to the appropriate audiences.